### ISSUE TITLE
one shot training results are extremely poor, I cannot reproduce the results of the paper

### ISSUE THREAD

**Issue by xiaqingsun** [7eu7d7/DreamArtist-sd-webui-extension#67]
When I train with a picture from the paper
<img width="131" alt="image" src="https://user-images.githubusercontent.com/106377665/218380529-3e1a0e31-7758-475d-b683-dc2a06963501.png">
my training results are as follows with the prompt "a painting of city, art by ch_fg4; ch_fg-neg" :
![image](https://user-images.githubusercontent.com/106377665/218381055-342115d3-e7d5-46a4-8255-20b72c7853e7.png)
![image](https://user-images.githubusercontent.com/106377665/218381167-a6fc6aff-d3af-403d-bebb-b39518080ef8.png)

but the results in the paper are like this:
<img width="651" alt="image" src="https://user-images.githubusercontent.com/106377665/218381398-a64f5eb7-61cf-4788-a334-282351ee2e37.png">

My training parameters are as follows:
![image](https://user-images.githubusercontent.com/106377665/218381519-f5bf4f41-0b62-4d11-803b-fe6464df9f2e.png)
![image](https://user-images.githubusercontent.com/106377665/218381804-1cee625b-752f-4880-b409-730e9288ffe5.png)
![image](https://user-images.githubusercontent.com/106377665/218381878-b2f27f88-e8f1-41a1-953a-1e35b08982f5.png)

### Is there something wrong with my training process causing this poor result？
Thanks！

**Comment by RainehDaze** [comment]
I would try it with fewer tokens and no initialisation text. Could also try lower CFG.

What might be an issue is the fact reconstruction is broken and there's been no update to fix it, though I don't know. I'll give it a shot later and try a few things, see if I can get anything vaguely similar.

**Comment by RainehDaze** [comment]
Hm, I wonder if it's the lack of reconstruction? I checked the paper, and it was pretty clear--three positive tokens, three negative tokens, 0.0025 learning rate, and a mysterious gamma that I think must be CFG as it was set as 5. 2-8k steps. So, I set it up to train that way, as well as a comparison with a much higher learning rate and EMA below 1 (and 6 negative tokens), and it's been kind of nonfunctional. The one with the really high learning rate sort of got something:

![image](https://user-images.githubusercontent.com/7266470/219001500-7ae2ea63-b71a-4058-8dd3-3686e7aab6b0.png)

But nothing on the paper results. Those all seem to generate generic cities, e.g.:
![rlinestyle2-2700](https://user-images.githubusercontent.com/7266470/219005177-8ef375b3-ba5f-45a4-ab9c-9da56f3f3e71.png)

**Comment by 7eu7d7** [comment]
set learning rate to 0.003 wolud be better, 0.0003 is too small

**Comment by 7eu7d7** [comment]
You can also try the improved version of DreamArtist++ with lora added for better results [HCP-Diffusion](https://github.com/7eu7d7/HCP-Diffusion)

**Comment by torridgristle** [comment]
This might be a shot in the dark, but if you still have the embeddings could you try using the negative embedding in the positive and the positive in the negative?

Recently I found that my embeds that previously looked bad actually worked pretty well in reverse and successfully captured elements of the training images much better than the non-reversed prompt. I don't know why this happened several times and I don't know if I made a bad edit to the script. As far as I can tell all I've done is changed it to not use the entire positive prompt as the negative prompt with only the trained embedding swapped for the negative version, and instead now it should be using only the negative embedding without the positive prompt from the prompt template.
